api/v1beta1/cluster_types.go

```diff
@@ -56,6 +56,59 @@ const (
 	// ClusterTopologyReconciledV1Beta2Condition is true if the topology controller is working properly.
 	// Note: This condition is added only if the Cluster is referencing a ClusterClass / defining a managed Topology.
 	ClusterTopologyReconciledV1Beta2Condition = "TopologyReconciled"
+
+	// ClusterTopologyReconcileSucceededV1Beta2Reason documents the reconciliation of a Cluster topology succeeded.
+	ClusterTopologyReconcileSucceededV1Beta2Reason = "TopologyReconcileSucceeded"
+
+	// ClusterTopologyReconciledFailedV1Beta2Reason documents the reconciliation of a Cluster topology
+	// failing due to an error.
+	ClusterTopologyReconciledFailedV1Beta2Reason = "TopologyReconcileFailed"
+
+	// ClusterTopologyReconciledControlPlaneUpgradePendingV1Beta2Reason documents reconciliation of a Cluster topology
+	// not yet completed because Control Plane is not yet updated to match the desired topology spec.
+	ClusterTopologyReconciledControlPlaneUpgradePendingV1Beta2Reason = "ControlPlaneUpgradePending"
```

Comment:
I think we should discuss which of those reason might impact cluster availability, and if not, improve the custom merge strategy in the Cluster controller. My first take is that TopologyReconcileFailed, ClusterClassNotReconciled are the only two reasons that should impact availability, everything else should be considered info

Reply:
Reframing my question. Do we want to only change the priority of the TopologyReconciled condition or do we have to do more?

Reply:
Implemented the priority thing. Seems okay to me

Reply:
Hm actually not sure if it's enough, maybe we need an overrideCondition, but PTAL. I added 4 test cases to the Available condition test, that should help to figure out what we want :)

Reply:
I think changing the priority is enough for now.
We can iterate when we go through the refinement phase next weeks / we see this condition together with all the other new conditions